wren-launcher/utils/rc.go

```diff
@@ -0,0 +1,169 @@
+// the wrenrc.go package is responsible for writing and reading rcfile
+// the rc file should located in ~/.wrenai/.wrenrc
+// should have  public methods "append" and "read"(read by key or all)
+// the structure of the rcfile is a env like file with key value pairs,
+// eg:
+//   "foo=bar"
+//   "bar=baz"
+
+package utils
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+)
+
+type WrenRC struct {
+	rcFileDir string
+}
+
+func (w *WrenRC) getWrenRcFilePath() string {
+	return path.Join(w.rcFileDir, ".wrenrc")
+}
+
+func (w *WrenRC) ensureRcFile() (string, error) {
+	// ensure folder created
+	err := os.MkdirAll(w.rcFileDir, os.ModePerm)
+	if err != nil {
+		return "", err
+	}
+
+	// ensure file created
+	rcFilePath := w.getWrenRcFilePath()
+	_, err = os.Stat(rcFilePath)
+	if os.IsNotExist(err) {
+		f, err := os.Create(rcFilePath)
+		if err != nil {
+			return "", err
+		}
+		f.Close()
+	}
+	return rcFilePath, nil
+}
+
+func (w *WrenRC) parseInto() (map[string]string, error) {
+	rcFilePath, err := w.ensureRcFile()
+	if err != nil {
+		return nil, err
+	}
+
+	f, err := os.Open(rcFilePath)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	// prepare a map to store the key value pairs
+	m := make(map[string]string)
+	// read the file line by line
+	r := bufio.NewReader(f)
+	lineno := 0
+	for {
```

Comment:
I refactored the for loop using spitN, I guess it will be more readable and robust
```
    // Read a line
		line, err := r.ReadString('\n')
		if err != nil {
			if err.Error() != "EOF" {
				return nil, err // return an error if it's not EOF
			}
			break
		}
		lineno++
		line = strings.Trim(line, " \t\v\r\n")

		// Skip empty lines and comments
		if line == "" || line[0] == '#' || line[0] == ';' {
			continue
		}

		// Split the line into key and value based on the '=' character
		parts := strings.SplitN(line, "=", 2)
		if len(parts) != 2 {
			return nil, fmt.Errorf("syntax error on line %d: no '=' character found", lineno)
		}

		// Trim spaces around key and value
		key := strings.TrimSpace(parts[0])
		value := strings.TrimSpace(parts[1])

		// Store the key-value pair in the map
		m[key] = value
		fmt.Println(key, value) // Optionally print the key and value
```

I test it with the following content
```
   <-- it is a empty line
foo=bar
   leadingPrefixK = leadingPrefixV
  valueWithEqual. =  path=to=foo
# commencted out
notcontain equal syntax

```
Result
```
 andyyen@Apple  ~/canner/WrenAI/wren-launcher  ↱ feature/telemetry ±  go run refactored.go
 foo bar
 leadingPrefixK leadingPrefixV
 valueWithEqual. path=to=foo
 syntax error on line 6: no '=' character found
 andyyen@Apple  ~/canner/WrenAI/wren-launcher  ↱ feature/telemetry ±  
```

Reply:
addressed in 77f1deb